plugin.video.bfi/resources/lib/cache.py

```diff
@@ -78,13 +82,19 @@ class Cache(object):
     https://docs.python.org/2/library/sqlite3.html
     https://tools.ietf.org/html/rfc7234
     """
+
     def __init__(self, name=None):
         self.connection = None
+        try:
```

Comment:
Hmm - not sure what xbmcvfs is can you link?

"Maybe check if the path exists before the try/except"

Not to contradict you - but AFAIK that is v.bad practice  - and it can lead to a nasty race condition.

You should never test for existence a path/socket/stream - then do something with it - as it could be created/deleted/destroyed in between the check and the next call - always better to try and fail gracefully as it it is bomb proof. 

Am happy to change it I am missing something?

Thanks again :)

Reply:
Ah - http://mirrors.kodi.tv/docs/python-docs/14.x-helix/xbmcvfs.html